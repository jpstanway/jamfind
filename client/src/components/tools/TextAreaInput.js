import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const TextAreaInput = ({
  label,
  name,
  placeholder,
  rows,
  value,
  onChange,
  disabled,
  error,
  info
}) => {
  return (
    <div className="form-group">
      {label && <label htmlFor={name}>{label}</label>}
      <textarea
        className={classnames("form-control form-control-lg", {
          "is-invalid": error
        })}
        name={name}
        id={name}
        placeholder={placeholder}
        rows={rows}
        value={value}
        onChange={onChange}
        disabled={disabled}
      />
      {info && <small className="form-text text-muted">{info}</small>}
      {error && <div className="invalid-feedback">{error}</div>}
    </div>
  );
};

TextAreaInput.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  rows: PropTypes.string,
  error: PropTypes.string,
  info: PropTypes.string
};

export default TextAreaInput;
