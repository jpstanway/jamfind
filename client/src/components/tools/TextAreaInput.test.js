import React from "react";
import { shallow } from "enzyme";
import TextAreaInput from "./TextAreaInput";

describe("<TextAreaInput />", () => {
  let mockOnChangeHandler, mockInputData;

  beforeEach(() => {
    mockOnChangeHandler = jest.fn();
    mockInputData = {
      label: "test",
      name: "test",
      placeholder: "type something",
      rows: "3",
      value: "",
      onChange: mockOnChangeHandler,
      disabled: "false",
      error: "test",
      info: "test",
    };
  });

  it("renders textarea input", () => {
    expect(shallow(<TextAreaInput {...mockInputData} />)).toMatchSnapshot();
  });

  it("can change value of textarea", () => {
    const wrapper = shallow(<TextAreaInput {...mockInputData} />);
    wrapper
      .find("textarea")
      .simulate("change", { target: { name: "test", value: "a" } });
    expect(mockOnChangeHandler.mock.calls.length).toBe(1);
  });
});
