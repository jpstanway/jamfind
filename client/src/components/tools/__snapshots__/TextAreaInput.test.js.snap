// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaInput /> renders textarea input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextAreaInput
    disabled="false"
    error="test"
    info="test"
    label="test"
    name="test"
    onChange={[MockFunction]}
    placeholder="type something"
    rows="3"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
        >
          test
        </label>,
        <textarea
          className="form-control form-control-lg is-invalid"
          disabled="false"
          id="test"
          name="test"
          onChange={[MockFunction]}
          placeholder="type something"
          rows="3"
          value=""
        />,
        <small
          className="form-text text-muted"
        >
          test
        </small>,
        <div
          className="invalid-feedback"
        >
          test
        </div>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control form-control-lg is-invalid",
          "disabled": "false",
          "id": "test",
          "name": "test",
          "onChange": [MockFunction],
          "placeholder": "type something",
          "rows": "3",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "form-text text-muted",
        },
        "ref": null,
        "rendered": "test",
        "type": "small",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "invalid-feedback",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
          >
            test
          </label>,
          <textarea
            className="form-control form-control-lg is-invalid"
            disabled="false"
            id="test"
            name="test"
            onChange={[MockFunction]}
            placeholder="type something"
            rows="3"
            value=""
          />,
          <small
            className="form-text text-muted"
          >
            test
          </small>,
          <div
            className="invalid-feedback"
          >
            test
          </div>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control form-control-lg is-invalid",
            "disabled": "false",
            "id": "test",
            "name": "test",
            "onChange": [MockFunction],
            "placeholder": "type something",
            "rows": "3",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "form-text text-muted",
          },
          "ref": null,
          "rendered": "test",
          "type": "small",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "invalid-feedback",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
